src/models.py

```diff
@@ -201,6 +209,89 @@ def _transform(self, datagen, validation_datagen=None):
         self.model.train()
 
 
+class ScoringLightGBM(LightGBM):
+    def __init__(self, model_params, training_params, train_size, target):
+        self.train_size = train_size
+        self.target = target
+        self.feature_names = []
+        self.estimator = None
+        super().__init__(model_params, training_params)
+
+    def fit(self, features, **kwargs):
+        df_features = []
+        for image_features in features:
+            for layer_features in image_features[1:]:
+                df_features.append(layer_features)
+        df_features = pd.concat(df_features)
+        train_data, val_data = train_test_split(df_features, train_size=self.train_size)
+        self.feature_names = list(df_features.columns.drop(self.target))
+        super().fit(X=train_data[self.feature_names],
+                    y=train_data[self.target],
+                    X_valid=val_data[self.feature_names],
+                    y_valid=val_data[self.target],
+                    feature_names=self.feature_names,
+                    categorical_features=[])
+        return self
+
+    def transform(self, features, **kwargs):
+        scores = []
```

Comment:
@jakubczakon I don't think it's possible. I mean, it is possible, but not very helpful, I think. That's because super().transform sits inside double `for` loop and it transforms data frame for each layer in each image.

Reply:
@apyskir typo here I meand prepare data and super().fit (as is)